src/components/batch/BatchSidebar/index.tsx

```diff
@@ -27,6 +27,14 @@ const BatchSidebar = ({ isOpen, onToggle }: { isOpen: boolean; onToggle: (open:
     batchTxs.forEach((item) => deleteTx(item.id))
   }, [deleteTx, batchTxs])
 
+  // Close confirmation flow when batch is empty
+  const shouldExitFlow = !!txFlow && batchTxs.length === 0
+  useEffect(() => {
+    if (shouldExitFlow) {
+      setTxFlow(undefined)
```

Comment:
Can we somehow couple this to the user action i.e. add it to the `deleteTx` call instead of having it in a `useEffect`?

Reply:
See https://github.com/safe-global/safe-wallet-web/pull/2453#discussion_r1307455063.

Reply:
Btw can there be a case when there’s some other tx flow open, not a batch modal?

Reply:
> Btw can there be a case when there’s some other tx flow open, not a batch modal?

I've updated this accordingly.